hazelcast/src/main/java/com/hazelcast/crdt/MutationDisallowedException.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.crdt;
+
+import com.hazelcast.core.HazelcastException;
+
+/**
+ * Exception that indicates that the state found on this replica disallows
+ * mutation. This may happen:
+ * <ul>
+ * <li>because the found state has already been migrated and is in the
+ * process of cleaning up</li>
+ * <li>a target replica is shutting down and in the process of replicating
+ * unreplicated CRDT state</li>
+ * <li>there are no data members to store the state</li>
```

Comment:
I think this exception is a little bit confusing. For instance, in the first and second cases, where a node does not perform an operation because it is no longer a crdt replica, we retry the operation in other nodes. In these two cases, this exception is handled internally. In the last case, there is no data node in the cluster so we propagate the exception to the caller. We already have an exception for this case: `NoDataMemberInClusterException`. Can we throw `NoDataMemberInClusterException` to the caller and keep this exception internal? WDYT?

Reply:
Didn't know about that exception. I will update it. OTOH the `MutationDisallowedException` may still be thrown if it is the last exception and the client has no other replicas it can try. Not sure how to avoid this one.

Reply:
Do we have a case where there is at least one data member but still cannot find a crdt replica?